mne/channels/_dig_montage_utils.py

```diff
@@ -0,0 +1,306 @@
+        coord_frame='head',
+    )
+
+
+def _read_dig_montage_egi(
+        fname,
+        _scaling,
+        _all_data_kwargs_are_none,
+):
+
+    if not _all_data_kwargs_are_none:
+        raise ValueError('hsp, hpi, elp, point_names, fif must all be '
+                         'None if egi is not None')
+    _check_fname(fname, overwrite='read', must_exist=True)
+
+    root = ElementTree.parse(fname).getroot()
+    ns = root.tag[root.tag.index('{'):root.tag.index('}') + 1]
+    sensors = root.find('%ssensorLayout/%ssensors' % (ns, ns))
+    fids = dict()
+    dig_ch_pos = dict()
+
+    fid_name_map = {'Nasion': 'nasion',
+                    'Right periauricular point': 'rpa',
+                    'Left periauricular point': 'lpa'}
+
+    for s in sensors:
+        name, number, kind = s[0].text, int(s[1].text), int(s[2].text)
+        coordinates = np.array([float(s[3].text), float(s[4].text),
+                                float(s[5].text)])
+
+        coordinates *= _scaling
+
+        # EEG Channels
+        if kind == 0:
+            dig_ch_pos['EEG %03d' % number] = coordinates
+        # Reference
+        elif kind == 1:
+            dig_ch_pos['EEG %03d' %
+                       (len(dig_ch_pos.keys()) + 1)] = coordinates
+            # XXX: we should do something with this (ref and eeg get mixed)
+
+        # Fiducials
+        elif kind == 2:
+            fid_name = fid_name_map[name]
+            fids[fid_name] = coordinates
+        # Unknown
+        else:
+            warn('Unknown sensor type %s detected. Skipping sensor...'
+                 'Proceed with caution!' % kind)
+
+    return Bunch(
+        # EGI stuff
+        nasion=fids['nasion'], lpa=fids['lpa'], rpa=fids['rpa'],
+        dig_ch_pos=dig_ch_pos, coord_frame='unknown',
+        # not EGI stuff
+        hsp=None, hpi=None, elp=None, point_names=None,
+    )
+
+
+def _foo_get_data_from_dig(dig):
+    # XXXX:
+    # This does something really similar to _read_dig_montage_fif but:
+    #   - does not check coord_frame
+    #   - does not do any operation that implies assumptions with the names
+
+    # Split up the dig points by category
+    hsp, hpi, elp = list(), list(), list()
+    fids, dig_ch_pos_location = dict(), list()
+
+    for d in dig:
+        if d['kind'] == FIFF.FIFFV_POINT_CARDINAL:
+            fids[_cardinal_ident_mapping[d['ident']]] = d['r']
+        elif d['kind'] == FIFF.FIFFV_POINT_HPI:
+            hpi.append(d['r'])
+            elp.append(d['r'])
+            # XXX: point_names.append('HPI%03d' % d['ident'])
+        elif d['kind'] == FIFF.FIFFV_POINT_EXTRA:
+            hsp.append(d['r'])
+        elif d['kind'] == FIFF.FIFFV_POINT_EEG:
+            # XXX: dig_ch_pos['EEG%03d' % d['ident']] = d['r']
+            dig_ch_pos_location.append(d['r'])
+
+    dig_coord_frames = set([d['coord_frame'] for d in dig])
+    assert len(dig_coord_frames) == 1, 'Only single coordinate frame in dig is supported' # noqa # XXX
```

Comment:
Hmm... this is part of the reason I wanted `Digitization` to win instead of just forcing everything into `DigMontage`. `Digitization` does not force everything into one coordinate frame. `Digitization` is also closer to what we store in FIF. So I'm not sold on this approach...

Reply:
Eventually we should relax this requirement, there is no problem with multiple coord frames. The critical thing is that the channels+extras do get converted when being set in `set_montage`, which means we should make `set_montage` call `dm.transform_to_head()` and say in the docs it will do it. This means that for polhemus data with UNKNOWN and MEG points to start, they get set in `info['dig']` in `HEAD` and `MEG` coord frames after calling `set_montage`.

Reply:
yes the idea was to relax it.